src/libraries/System.Threading.RateLimiting/src/System/Threading/RateLimiting/MetadataName.T.cs

```diff
@@ -0,0 +1,81 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics.CodeAnalysis;
+
+namespace System.Threading.RateLimiting
+{
+    /// <summary>
+    /// A strongly-typed name of metadata that can be stored in a <see cref="RateLimitLease"/>.
+    /// </summary>
+    /// <typeparam name="T">The type the metadata will be.</typeparam>
+    public sealed class MetadataName<T> : IEquatable<MetadataName<T>>
+    {
+        private readonly string _name;
+
+        /// <summary>
+        /// Constructs a <see cref="MetadataName{T}"/> object with the given name.
+        /// </summary>
+        /// <param name="name">The name of the <see cref="MetadataName"/> object.</param>
+        public MetadataName(string name)
+        {
```

Comment:
Should this throw if name is null?  Then you can remove the null checks from ToString and GetHashCode.